docs/book/src/topics/custom-vm-extensions.md

```diff
@@ -0,0 +1,68 @@
+# Custom VM Extensions
+
+## Overview
```

Comment:
Thanks for adding this doc, looks good 💯 

Can you please add it to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7d8dc8583b1c8db99a84768e5ed7f2f140983a05/docs/book/src/SUMMARY.md?plain=1 under Topics so it shows up in the book Table Of Content?

Reply:
Got it! I'll add it under the other "custom" topics. I'm not sure if there is a specific ordering of topics, but I felt like that would be the best place to add it!

Reply:
I thought so too, but it seems like a few are out of order. "Custom Images" is under "Custom Private DNS Zone name", and there are a few others like "SSH Access to nodes". Should we standardize how we order the topics?

Reply:
I'm guessing "Custom Private DNS Zone" is ordered by file name "custom-dns" (d comes before i) and "SSH Access to nodes" was likely just added at the end by someone who didn't realize the list was in order. Feel free to reorganize/rename however you see fit, the ordered list is becoming less practical now that we have so many topics, maybe we need sub-sections now. This can be a separate PR so it doesn't block this one.